applications/roloi-xcm-payment-automation.md

```diff
@@ -0,0 +1,199 @@
+# Roloi - XCM Payment Automation
+
+- **Team Name:** NeoPower Digital
+- **Payment Address:** 1dz667uQX199rHyj6tizmL6EJe4AZxjH1RhnyrT1QuQ4pfS (Polkadot - USDT)
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 3
+
+## Project Overview :page_facing_up:
+
+### Overview
+
+We are [NeoPower](https://twitter.com/NeoPowerDigital), a Web3 software company. We are currently building a payment automation platform called [Roloi](https://twitter.com/RoloiMoney).
+
+Last year we received a [W3F Grant](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/roloi_milestone_1.md) for the migration of our Streaming Smart Contract from Cosmos to Polkadot. We talk about this migration journey in a [Medium Article](https://medium.com/@RoloiMoney/cosmwasm-to-ink-beyond-the-cosmos-e4920604f9cb).
+
+Roloi is based on three automation pillars:
+- Token streaming
+- Recurring payments
+- Conditional payments
+
+As a next step for Roloi, we are going to build the **recurring payments feature**. 
+
+Roloi can achieve this by avoiding off-chain mechanisms thanks to [OAK Network](https://oak.tech/): the automation parachain.
+
+### Project Details
+
+#### Goal
+
+Automate ink! smart contract transactions in a recurring way.
+
+#### How to achieve it
+
+We need to notify our Smart Contract deployed on **Shiden** (Astar) whenever a **scheduled transaction** should be executed.
+
+**XCM** enables us to connect parachains. This way, we can schedule transactions on **Shiden** (Astar) with the assistance of **Turing** (OAK). 
+
+#### Technical details
+
+Creating products using XCM implies a complex journey today. Here is an example between Astar and OAK:
+
+![W3F Grant - XCM Payment Automation](https://i.imgur.com/lrjifU5.png)
```

Comment:
Can you provide another image that would show how the workflow would be with smart contract? Who would be paying the fees for the recurring transactions, and how? Would the smart contract manage the fees per user?

Reply:
The workflow diagram generically shows the communication between an origin chain and OAK to enable recurring tasks, and the extrinsic can be any type of task. In our case, will be a smart contract transaction. Internally, the contract will validate the execution and trigger the configured transfer.

As for the fees, the idea is for users to top up the proxy accounts based on the defined recurrence settings. 
On the UI side, we can simulate and estimate the fees of XCM and smart contract transactions to cover an approximate number of recurring repetitions. Then we’ll suggest to the user a concrete number of tokens to transfer to the proxy account to cover fees. 
We’ll also use these estimation calculations on the UI to alert the user when they are running out of funds to pay the fees, so they can recharge the proxy account.